src/consts/models.tsx

```diff
@@ -116,3 +127,109 @@ export interface Filters {
   toAmount?: number;
   transactionType?: string;
 }
+
+export type MainTabNavigatorParamList = {
+  [Route.Dashboard]: undefined;
+  [Route.ContactList]: undefined;
+  [Route.Settings]: undefined;
+};
+
+export type RootStackParamList = {
+  [Route.MainCardStackNavigator]: undefined;
+  [Route.ImportWalletQRCode]: undefined;
+  [Route.ActionSheet]: { wallets: Wallet[]; selectedIndex: number; onPress: (index: number) => void };
+  [Route.UnlockTransactionNavaigator]: undefined;
+  [Route.PasswordNavigator]: undefined;
+  [Route.EditTextNavigator]: {
+    title: string;
+    onSave: (value: string) => void;
+    label: string;
+    header?: React.ReactNode;
+    value?: string;
+    validate?: (value: string) => string | undefined;
+    keyboardType?: KeyboardType;
+  };
+  [Route.Message]: {
+    title: string;
+    source: FastImageSource;
+    description: string;
+    buttonProps?: ButtonProps;
+    imageStyle?: StyleProp<ViewStyle>;
+    asyncTask?: () => void;
+  };
+  [Route.ExportWallet]: { wallet: Wallet };
+  [Route.ExportWalletXpub]: { wallet: Wallet };
+  [Route.DeleteWallet]: { wallet: Wallet };
+  [Route.DeleteContact]: { contact: Contact };
+  [Route.MainCardStackNavigator]: undefined;
+  [Route.SendTransactionDetails]: {
+    fee: number;
+    recipients: any;
+    tx: any;
+    satoshiPerByte: any;
+    wallet: Wallet;
+  };
+};
+
+export type UnlockTransactionNavigatorParamList = {
+  [Route.UnlockTransaction]: { onSuccess: () => void };
+};
+
+export type PasswordNavigatorParamList = {
+  [Route.CreateTransactionPassword]: undefined;
+  [Route.ConfirmTransactionPassword]: { setPassword: string };
+};
+
+export type EditTextNavigatorParamList = {
+  [Route.EditText]: {
+    title: string;
+    onSave: (value: string) => void;
+    label: string;
+    header?: React.ReactNode;
+    value?: string;
+    validate?: (value: string) => string | undefined;
+    keyboardType?: KeyboardType;
+  };
+};
+
+export type MainCardStackNavigatorParamList = {
+  [Route.MainCardStackNavigator]: undefined;
+  [Route.CreateWallet]: undefined;
+  [Route.ImportWallet]: { appSettings: AppSettingsState; loadWallets: () => Promise<WalletsActionType> };
```

Comment:
I'm not really a big fan of such formatting with prettier. I believe we should talk about always making larger objects like
```
{
  appSettings: AppSettingsState; 
  loadWallets: () => Promise<WalletsActionType>;
};

There's a rule for that I guess, just need to decide if we want to apply it or not :)